dcpy/utils/s3.py

```diff
@@ -367,14 +378,19 @@ def delete(bucket: str, path: str) -> None:
     client_.delete_object(Bucket=bucket, Key=path)
 
 
-def get_filenames(bucket: str, prefix: str) -> set[str]:
+def get_suffixes(bucket: str, prefix: str) -> set[str]:
     return {
-        obj["Key"].split("/")[-1]
+        obj["Key"].removeprefix(prefix)
         for obj in list_objects(bucket, prefix)
-        if obj["Key"].split("/")[-1] != ""
+        if obj["Key"].removeprefix(prefix) != ""
     }
 
 
+def get_filenames(bucket: str, prefix: str) -> set[str]:
```

Comment:
Bit of a change in behavior from the old `get_filenames` (which I've replaced with this and the above `get_suffixes`, which is a bit more general). Before, we were getting all filenames without any directories included regardless of level. So if we had
```
/folder1
    /folder2
        file1.txt
    file2.txt
```
And we called this on `folder1` we'd get `{ "file1.txt", "file2.txt" }`. More confusingly, if they were named the same, we'd only get back one of them.

Now, we'd get back `{ "folder2/file1.txt", "file2.txt" }` which seems much more intuitive